### ISSUE TITLE
Hakkar Event Nightmare Suppressor

### ISSUE THREAD

**Issue by TheTrueAnimal** [cmangos/issues#469]
Nightmare suppressors are part of the hakkar event in sunken temple. They should spawn (works correctly), run to the middle of the room and start to cast a spell. If the players don't make them stop casting (by attacking him) the event gets interruppted and the group fails. At the moment the npcs spawn at the correct moment but they don't do anything.

NPC: http://www.wowhead.com/npc=8497
Spell: http://de.wowhead.com/spell=12623
Video: https://www.youtube.com/watch?v=44e3dJysjfk
https://www.youtube.com/watch?v=sAmRO6qv9_Y(from 2:00 on)
Source: http://www.wowhead.com/npc=8497#comments
http://hordeguides.de/Instanz/Classic/Boss/8443/Avatar-von-Hakkar/index.htm (german)

**Comment by xfurry** [comment]
@cala this issue is because of missing DB data.

https://github.com/cmangos/mangos-classic/blob/master/src/scriptdev2/scripts/eastern_kingdoms/sunken_temple/instance_sunken_temple.cpp#L427
This code spawns the creature.
The DB should contain the dbscript for the creature movement and spell casting. I think this was probably lost in the old UDB forum.
I know this one worked for sure, because I developed the event myself.

**Comment by cala** [comment]
@xfurry I indeed added data in Classic DB last year or the year before to make the NPCs move towards the boss and cast the spell, including the spell script target. This works pretty well (I can't tell if I took the data from UDB or remade them myself).

The only thing I did not manage to make work is the despawn of the boss if the spell is cast on it too long (a few seconds or dozen of seconds), making the whole encounter fail.
